api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -525,47 +520,33 @@ private void formatTimestampValues(List<DataSetPreviewValueList> valuePreviewLis
       String workspaceNamespace, String workspaceId, MarkDataSetRequest markDataSetRequest) {
     workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
         workspaceNamespace, workspaceId, WorkspaceAccessLevel.WRITER);
-    List<DbDataset> dbDataSetList = new ArrayList<>();
-    if (ResourceType.COHORT.equals(markDataSetRequest.getResourceType())) {
-      dbDataSetList = dataSetDao.findDataSetsByCohortIds(markDataSetRequest.getId());
-    } else if (ResourceType.CONCEPT_SET.equals(markDataSetRequest.getResourceType())) {
-      dbDataSetList = dataSetDao.findDataSetsByConceptSetIds(markDataSetRequest.getId());
-    }
-    dbDataSetList =
-        dbDataSetList.stream()
-            .map(
-                dataSet -> {
-                  dataSet.setInvalid(true);
-                  return dataSet;
-                })
-            .collect(Collectors.toList());
-    try {
-      dataSetDao.save(dbDataSetList);
-    } catch (OptimisticLockException e) {
-      throw new ConflictException("Failed due to concurrent data set modification");
-    }
-
+    dataSetService.markDirty(markDataSetRequest.getResourceType(), markDataSetRequest.getId());
     return ResponseEntity.ok(true);
   }
 
   @Override
   public ResponseEntity<EmptyResponse> deleteDataSet(
       String workspaceNamespace, String workspaceId, Long dataSetId) {
-    DbDataset dataSet =
-        getDbDataSet(workspaceNamespace, workspaceId, dataSetId, WorkspaceAccessLevel.WRITER);
-    dataSetDao.delete(dataSet.getDataSetId());
+    DbWorkspace workspace =
+        workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
```

Comment:
I'd add a method to WorkspaceService that returns Optional<DbWorkspace> so you know to handle the case where it's not found separately.